src/Serilog/ReusableStringWriter.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace Serilog;
+
+/// <summary>
+/// Class that provides reusable StringWriters to reduce memory allocations
+/// </summary>
+public class ReusableStringWriter: StringWriter
+{
+    static readonly object _poolLock = new();
+    static readonly Dictionary<IFormatProvider, Stack<ReusableStringWriter>> _writerPools = new();
+
+    /// <summary>
+    /// Gets already created StringWriter if there is one available or creates a new one.
+    /// </summary>
+    /// <param name="formatProvider"></param>
+    public static StringWriter GetOrCreate(IFormatProvider? formatProvider = null)
```

Comment:
There's a small risk here that user code could trigger a space leak by supplying unique format providers - would need to consider mitigating this.

Reply:
The first think I would do - do not use pooling when formatProvider is not null.